pkg/menu/menu.go

```diff
@@ -71,3 +72,29 @@ func GetInput(introwords string, location int, wid int, ht int, isValid func(str
 	uiEvents := ui.PollEvents()
 	return processInput(input, warning, isValid, uiEvents)
 }
+
+// create a new ui window and display an input box.
+func NewCustomInputWindow(introwords string, wid int, ht int, isValid func(string) (string, bool)) (string, error) {
+	uiEvents := ui.PollEvents()
+	return internalNewInputWindow(introwords, wid, ht, isValid, uiEvents)
+}
+
+func internalNewInputWindow(introwords string, wid int, ht int, isValid func(string) (string, bool), uiEvents <-chan ui.Event) (string, error) {
+	if err := ui.Init(); err != nil {
+		return "", fmt.Errorf("Failed to initialize termui: %v", err)
+	}
+	defer ui.Close()
+
+	location := 0
+	intro := newParagraph(introwords, false, location, len(introwords)+4, 3)
+	location += 2
+	input := newParagraph("", true, location, wid, ht+2)
+	location += ht + 2
+	warning := newParagraph("", false, location, wid, 3)
+
+	ui.Render(intro)
+	ui.Render(input)
+	ui.Render(warning)
+
+	return processInput(input, warning, isValid, uiEvents)
```

Comment:
Ok I see what you mean now, I didn't realize getInput was only using internally by other menu pkg functions; in that case, what about making the following slight change that kind of goes back to what you used to do

```
func processInput(introwords string, wid int, ht int, isValid func(string) (string, bool), uiEvents <-chan ui.Event) {
   // set up the *widgets.Paragraph like you used to before
   // rest of processInput function body
}
```
so that you don't have to set up the new paragraphs every time. Then all NewCustomInputWindow would need to do is do the ui.Init and then call processInput.

Reply:
but I will unable to test warning if I remove the warning parameter